jmix-dynattr/dynattr-flowui/src/main/java/io/jmix/dynattrflowui/view/category/CategoryDetailView.java

```diff
@@ -0,0 +1,507 @@
+        setGridActionsEnabled(!categoryAttrsGrid.getSelectedItems().isEmpty());
+    }
+
+    private void initActionsVisibleListener() {
+        categoryAttrsGrid.addSelectionListener(e -> setGridActionsEnabled(!e.getAllSelectedItems().isEmpty()));
+    }
+
+    private void setGridActionsEnabled(boolean enabled) {
+        List.of(editAction, removeAction, moveUpAction, moveDownAction)
+                .forEach(button -> button.setEnabled(enabled));
+    }
+
+    protected void setupFieldsLock() {
+        CrudEntityContext crudEntityContext = new CrudEntityContext(categoryDc.getEntityMetaClass());
+        accessManager.applyRegisteredConstraints(crudEntityContext);
+        if (!crudEntityContext.isUpdatePermitted()) {
+            entityTypeField.setEnabled(false);
+        }
+    }
+
+    @Subscribe("entityTypeField")
+    protected void onEntityTypeFieldValueChange(HasValue.ValueChangeEvent<MetaClass> event) {
+        if (event.getValue() != null) {
+            getEditedEntity().setEntityType(event.getValue().getName());
+        }
+    }
+
+    @Subscribe("isDefaultField")
+    protected void onIsDefaultFieldValueChange(HasValue.ValueChangeEvent<Boolean> event) {
+        if (Boolean.TRUE.equals(event.getValue())) {
+            FetchPlan fetchPlan = fetchPlans.builder(Category.class)
+                    .add("isDefault")
+                    .build();
+            LoadContext<Category> loadContext = new LoadContext<Category>(metadata.getClass(Category.class))
+                    .setFetchPlan(fetchPlan);
+            Category category = getEditedEntity();
+            loadContext.setQueryString("select c from dynat_Category c where c.entityType = :entityType and not c.id = :id")
+                    .setParameter("entityType", category.getEntityType())
+                    .setParameter("id", category.getId());
+            List<Category> foundCategories = dataManager.loadList(loadContext);
+            foundCategories.forEach(item -> item.setIsDefault(false));
+
+            dataManager.save(new SaveContext().saving(foundCategories));
+        }
+    }
+
+    protected void initEntityTypeField() {
+        Map<MetaClass, String> options = new HashMap<>(); //the map sorts metaclasses by the string key
+        for (MetaClass metaClass : metadataTools.getAllJpaEntityMetaClasses()) {
+            if (metadataTools.hasCompositePrimaryKey(metaClass) && !metadataTools.hasUuid(metaClass)) {
+                continue;
+            }
+            options.put(metaClass, messageTools.getDetailedEntityCaption(metaClass));
+        }
+        entityTypeField.setItemLabelGenerator(options::get);
+        //noinspection unchecked
+        entityTypeField.setItems(DataProviderUtils.createCallbackDataProvider(options.keySet().stream()
+                .sorted(Comparator.comparing(MetadataObject::getName))
+                .toList()));
+
+        entityTypeField.addValueChangeListener(e -> getEditedEntity().setEntityType(e.getValue().getName()));
```

Comment:
You already have `@Subscribe("entityTypeField")` on `ValueChangeEvent`. Also, you change field value in its own value change listener based on value that you set to the edited entity one line before.